torch/nn/utils/prune.py

```diff
@@ -72,10 +61,23 @@ def apply_mask(self, module):
         ), "Module {} has to be pruned".format(
             module
         )  # this gets set in apply()
-        mask = getattr(module, self._tensor_name + "_mask")
-        orig = getattr(module, self._tensor_name + "_orig")
-        pruned_tensor = mask.to(dtype=orig.dtype) * orig
-        return pruned_tensor
+        mask = getattr(self._module, self._tensor_name + "_mask")
```

Comment:
Yeah, this is the part from the design of the Parametrizations I was the least happy about. It does form a circular reference, which is almost never desired, but given that the Parametrizations when in `auto` mode are stored in the hooks of the tensor, they do not have access to anything when the hook is invoked.
A partial solution to this would be to use a weak reference (think that I should have done in retrospective, will add this in the next commit) to store the reference to the module. This makes a lot of sense, since parametrizations always live tied to a module.

Reply:
I tried the weak references approach, but it poses problems when doing `torch.save`. It throws  in `serialization.py` the following
```
>       pickler.dump(obj)
E       TypeError: cannot pickle 'weakref' object
```
I'm not sure how to go about this one...